server/src/main/java/org/opensearch/index/seqno/GlobalCheckpointSyncAction.java

```diff
@@ -98,7 +100,10 @@ public GlobalCheckpointSyncAction(
     public void updateGlobalCheckpointForShard(final ShardId shardId) {
         final ThreadContext threadContext = threadPool.getThreadContext();
         try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
-            threadContext.markAsSystemContext();
+            AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
```

Comment:
We could extract this check into `ThreadContextAccess` utility class (similarly to `SocketAccess` that we use in many plugins). Not a big deal but much less noise / code required

Reply:
Sounds like a good idea. I will make that change.

Reply:
Pushed a change to introduce `ThreadContextAccess` that is modeled after `SocketAccess`.